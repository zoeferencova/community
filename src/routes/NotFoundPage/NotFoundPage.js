import React from "react";
import { useNavigate } from "react-router-dom";
import { ButtonLight } from "../../components/Utils/Utils"
import { PropTypes } from 'prop-types';

import styles from "./NotFoundPage.module.css"

const NotFoundPage = ({ isLoggedIn }) => {
    const navigate = useNavigate();

    // Redirects user to HomePage or LandingPage based on whether or not the user is logged in
    const redirect = () => isLoggedIn ? navigate("/home") : navigate("/");

    return (
        <>
            <div className={(window.location.pathname === "/location" && isLoggedIn) ? styles.hide : styles.notFound}>
                <img src={require("../../images/not-found.png")} alt="Not found error" className={styles.image}></img>
                <h1 className={styles.title}>Sorry, that page couldn't be found</h1>
                <ButtonLight onClick={redirect}>Take me home</ButtonLight>
            </div>
        </>

    )
}

export default NotFoundPage;

NotFoundPage.propTypes = {
    isLoggedIn: PropTypes.bool
}