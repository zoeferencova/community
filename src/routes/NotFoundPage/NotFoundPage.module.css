.hide {
    display: none;
}

.notFound {
    display: flex;
    align-items: center;
    height: 100%;
    flex-direction: column;
}

.image {
    height: 350px;
}

.title {
    font-size: 25px;
    margin-top: 0px;
    margin-bottom: 40px;
    max-width: 80%;
}

.notFound button {
    margin-bottom: 40px;
}