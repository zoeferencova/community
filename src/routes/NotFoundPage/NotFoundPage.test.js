import React from "react";
import { shallow } from "enzyme";
import NotFoundPage from "./NotFoundPage";


describe("NotFoundPage", () => {
  it("renders NotFoundPage component", () => {
    const wrapper = shallow(<NotFoundPage />);

    expect(wrapper).toMatchSnapshot();
  });
});




