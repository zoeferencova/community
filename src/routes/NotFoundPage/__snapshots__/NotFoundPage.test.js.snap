// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders NotFoundPage component 1`] = `
<Fragment>
  <h1
    className=""
  >
    NotFoundPage
  </h1>
</Fragment>
`;
